### PR TITLE
[v7r1] initial support for mysql8

### PR DESCRIPTION
**Pull request by ltomassetti**
in mysql8 is not possible to create users with the GRANT statement anymore:
- fix GRANT permission statement to Dirac user(s) in ComponentInstaller.py by removing IDENTIFIED BY part
in mysql8 there are several more reserved word, among which 'SYSTEM'.
- fix query construction in ComponentInstaller.py by quoting with backtick the column names

BEGINRELEASENOTES

*Core
NEW: initial support for mysql8

ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
Ignore the "pylintPY3K" failure, it's fixed in #4163

File: FrameworkSystem/DB/InstalledComponentsDB.py [hunk 1]
```diff
@@ -429,10 +429,10 @@ def __filterFields(self, session, table, matchFields=None):
         actualKey = key.replace('.smaller', '')
 
       if matchFields[key] is None:
-        sql = '%s IS NULL' % (actualKey)
+        sql = '`%s` IS NULL' % (actualKey)
       elif isinstance(matchFields[key], list):
         if len(matchFields[key]) > 0 and None not in matchFields[key]:
-          sql = '%s IN ( ' % (actualKey)
+          sql = '`%s` IN ( ' % (actualKey)
```
Comment (chaen) [inline FrameworkSystem/DB/InstalledComponentsDB.py]: Aren't we better off just moving all these trivial DBs to SQLAlchemy once for all ?

Comment (fstagni) [reply FrameworkSystem/DB/InstalledComponentsDB.py]: ...this specific DB is already written in SQLAlchemy!

Comment (chaen) [reply FrameworkSystem/DB/InstalledComponentsDB.py]: but why do we have to manipulate query string then ?

Comment (ltomassetti) [reply FrameworkSystem/DB/InstalledComponentsDB.py]: MySQL v8 introduces several 'new' reserved word (`system` for instance). These can be used as column-name only if back-tick-quoted in queries. Quoting every column-name is maybe too much, but safe. Anyway, column-names (and table-names) were already quoted in the `SELECT` and `FROM` clauses. They weren't in the `WHERE` one.

Comment (chaen) [reply FrameworkSystem/DB/InstalledComponentsDB.py]: This i understand. I am just wondering why in a SqlAlchemy powered DB, we need to write queries at all ? SQLAlechemy always escapes column names

Comment (fstagni) [reply FrameworkSystem/DB/InstalledComponentsDB.py]: The strange thing is that in this DB there's one method where an sql statement is constructed. This is definitely not ideal. We can think or re-write the `__filterFields` method or accept this PR as-is and maybe re-write later.

**Comment by fstagni** [comment]
I changed the base to `rel-v7r1`, and it added some changes to releases.cfg, that will probably be innocuous here. Anyway, @atsareg consider them.

**Pull request merged**